packages/swarm-components/src/Radio.jsx

```diff
@@ -1,27 +1,60 @@
 // @flow
 import * as React from 'react';
-import Icon from './Icon';
 
 export type Props = {
-  checked: boolean,
+  /**
+   * Whether the radio button is checked.
+   */
+  checked?: boolean,
+  /**
+   * Whether the input should be interactive.
+   */
   disabled?: boolean,
+  /**
+   * A unique identifier for the input.
+   */
   id?: string,
+  /**
+   * Used to associate a group of radio buttons.
+   * Only one radio button in a group can be selected.
+   */
+  name: string,
+  /**
+   * Value of the input.
+   */
   value: string,
+  /**
+   * Label for the input.
+   */
   label?: string,
-  name: string,
+  /**
+   * Use children as an alternative to the `label` prop for more complex input labels.
+   */
+  children: React.ReactElement<*>,
 };
 
 // Can not inline css vars as color to Icon. Gray 5 icon fill on disbaled
 const Radio = (props: Props): React.ReactElement<'label'> => {
-  const { checked, label, id = 'a', disabled, value, children, ...rest } = props;
+  const {
+    checked,
+    label,
+    id,
+    disabled,
+    value,
+    children,
+    name,
+    ...other
+  } = props;
+
   return (
-    <label data-swarm-radio={disabled ? 'disabled' : 'default'} for={id} disabled={disabled} {...rest} >
+    <label data-swarm-radio={disabled ? 'disabled' : 'default'} htmlFor={id || name} disabled={disabled} {...other} >
       <span data-swarm-radio-field={checked ? 'checked': 'unchecked'} tabIndex='0' role="checkbox" aria-checked={checked}>
         {checked && <span data-swarm-radio-dot/>}
       </span>
-      <input type='radio' id={id} checked={checked} disabled={disabled} value={value} />
+      <input type='radio' id={id || name} name={name} checked={checked} disabled={disabled} value={value} />
```

Comment:
Just realized that we use name as an ID which is not correct if I understand it properly, as IDs are unique per page while names are not and actually used to group radio buttons together.

Reply:
I'll fix this.